packages/cli/src/utils.ts

```diff
@@ -197,6 +203,41 @@ export function tsTemplate(
   }\n//# sourceMappingURL=${filename}.map\n`;
 }
 
+function isValidJsIdentifier(name: string) {
+  return /^(?!\d)[\w$]+$/.test(name);
+}
+
+export function mjsTemplate(exports: string[], relativePath: string) {
+  const escapedPath = JSON.stringify(relativePath);
+  return `import * as _ns from ${escapedPath};\n${exports
+    .map((name, i) => {
+      if (name === "default") {
+        return `export default _ns.default.default;`;
+      }
+      if (!isValidJsIdentifier(name)) {
+        const escapedName = JSON.stringify(name);
+        return `var _export${i} = _ns[${escapedName}];\nexport { _export${i} as ${escapedName} };`;
+      }
+      return `export var ${name} = _ns.${name};`;
+    })
+    .join("\n")}\n`;
+}
+
+export function dmtsTemplate(exports: string[], relativePath: string) {
+  const escapedPath = JSON.stringify(`${relativePath}.js`);
+  return `import * as _ns from ${escapedPath};\n${exports
+    .map((name, i) => {
+      if (name === "default") {
+        return `declare const _def: typeof _ns.default.default;\nexport default _def;`;
+      }
+      if (!isValidJsIdentifier(name)) {
+        throw new Error("TypeScript does not support non-identifier exports");
+      }
```

Comment:
I couldn't make it work with TS, it doesn't like this valid JS:
```js
export let a = 1;
export { a as "test" }
// Identifier expected.ts(1003)
// Cannot find module 'test' or its corresponding type declarations.ts(2307)
```

Reply:
Could we just emit the string for invalid identifiers so that when TS adds support, it'll just work?

Reply:
It's no longer a concern since we just emit `export * from`